WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.kt

```diff
@@ -755,6 +761,59 @@ class MySiteFragment : Fragment(),
         }
     }
 
+    private fun isWPStoriesMediaBrowserTypeResult(data: Intent): Boolean {
+        if (data.hasExtra(MediaBrowserActivity.ARG_BROWSER_TYPE)) {
+            val browserType = data.getSerializableExtra(MediaBrowserActivity.ARG_BROWSER_TYPE)
+            return browserType == MediaBrowserType.WP_STORIES_MEDIA_PICKER
+        }
+        return false
+    }
+
+    /*
+        return true if MediaPickerResult was handled
+     */
+    private fun handleMediaPickerResultForStories(data: Intent): Boolean {
+        if (data.getBooleanExtra(PhotoPickerActivity.EXTRA_LAUNCH_WPSTORIES_CAMERA_REQUESTED, false)) {
+            ActivityLauncher.addNewStoryForResult(
+                    activity,
+                    selectedSite,
+                    PagePostCreationSourcesDetail.STORY_FROM_MY_SITE
+            )
+            return true
+        } else if (isWPStoriesMediaBrowserTypeResult(data)) {
+            if (data.hasExtra(MediaBrowserActivity.RESULT_IDS)) {
+                ActivityLauncher.addNewStoryWithMediaIdsForResult(
+                        activity,
+                        selectedSite,
+                        PagePostCreationSourcesDetail.STORY_FROM_MY_SITE,
+                        data.getLongArrayExtra(
+                                MediaBrowserActivity.RESULT_IDS
+                        )
+                )
+                return true
+            } else {
+                val mediaUriStringsArray = data.getStringArrayExtra(
```

Comment:
this will add media URIs (for example, when picking media from the device, or free photo library)